src/SBInterpolatedImage.cpp

```diff
@@ -900,6 +782,198 @@ namespace galsim {
+                which works out to
+                2 i ( kx L cos(kx L/2) - 2 sin(kx L/2)) sin (ky L/2) / kx^2 / ky.
+                Noting that kx L/2 = ikx pi, the cosines are -1^ikx and the sines are 0.
+                Of course, lim kx->0 sin(kx)/kx is 1 though, so that term survives.  Algebra
+                eventually reduces the above expression to what's in the code below.
+             */
+            double xsum(0.0), ysum(0.0);
+            for (int iky = -Ninitial/2; iky < Ninitial/2; iky++) {
+                if (iky == 0) continue;
+                ysum += std::pow(-1.0, iky) / iky * _ktab->kval(0, iky).imag();
+            }
+            for (int ikx = -Ninitial/2; ikx < Ninitial/2; ikx++) {
+                if (ikx == 0) continue;
+                xsum += std::pow(-1.0, ikx) / ikx * _ktab->kval(ikx, 0).imag();
+            }
+            xcentroid = xsum/_dk/flux;
+            ycentroid = ysum/_dk/flux;
+        }
+        return Position<double>(xcentroid, ycentroid);
+    }
+
+    std::string SBInterpolatedKImage::SBInterpolatedKImageImpl::repr() const
+    {
+        std::ostringstream oss(" ");
+        oss.precision(std::numeric_limits<double>::digits10 + 4);
+        oss << "galsim._galsim.SBInterpolatedKImage(";
+
+        oss << "galsim._galsim.ConstImageViewD(array([";
+        ConstImageView<double> reim = getRealKImage();
+        int N = _ktab->getN();
+        for (int y = 0; y<N; ++y) {
+            if (y > 0) oss <<",";
+            BaseImage<double>::const_iterator it = reim.rowBegin(y);
+            oss << "[" << *it++;
+            for (; it != reim.rowEnd(y); ++it) oss << "," << *it;
+            oss << "]";
+        }
+        oss<<"],dtype=float)), ";
+
+        oss << "galsim._galsim.ConstImageViewD(array([";
+        ConstImageView<double> imim = getImagKImage();
+        for (int y = 0; y<N; ++y) {
+            if (y > 0) oss <<",";
+            BaseImage<double>::const_iterator it = imim.rowBegin(y);
+            oss << "[" << *it++;
+            for (; it != imim.rowEnd(y); ++it) oss << "," << *it;
+            oss << "]";
+        }
+        oss<<"],dtype=float)), ";
+
+        oss << _ktab->getDk() << ", " << stepK() << ", ";
+        boost::shared_ptr<Interpolant> kinterp = getKInterp();
+        oss << "galsim.Interpolant('"<<kinterp->makeStr()<<"', "<<kinterp->getTolerance()<<"), "
+            << "galsim.GSParams("<<*gsparams<<"))";
+        return oss.str();
+    }
+
+    ConstImageView<double> SBInterpolatedKImage::SBInterpolatedKImageImpl::getRealKImage() const
+    {
+        int N = _ktab->getN();
+        tmv::Matrix<std::complex<double> > val(N,N);
+        int Nko2 = Nk/2;
+        fillKValue(val.view(), -Nko2, 1, Nko2, -Nko2, 1, Nko2);
+        ImageAlloc<double> im(N, N);
+        tmv::MatrixView<double> m(im.getData(), N, N, 1, im.getStride(), tmv::NonConj);
+        m += val.realPart();
+        return im;
+    }
```

Comment:
This isn't what we want here.  The only use of these two functions is to be able to serialize the SBInterpolatedKImage more efficiently than the version we get by using `repr`.  Calling `fillKValue` is not very efficient.  

For SBInterpolatedImage, it was fine to take the view of the XTable, since that has the right shape and it doesn't actually require copying any values to make the ImageView, so it is efficient to pass that back and reasonably efficient to remake the SBInterpolatedImage from that.

Here, KTable stores things in a different order than the input images, so getting a view of the real and imag parts wouldn't be the right thing to input back as a the real and imag images in the constructor. Which is why you implemented these functions this way.  But if we're going to do all this, it's probably better to just keep the originals around and use them again.

However, I think a better option would be to make an alternate version of the constructor that would accept the raw KValue data as input and efficiently recreate the KTable.  I'd suggest adding the centroid values to the arg list so if they have been calculated, then it would avoid recalculating those as well.  (You'd also need the bool flag of whether they are calculated or not yet.)

Then instead of these two functions, we would have one function that returns the KValue data packaged up as an Image, but not really being any kind of useful image.

```
template <typename T>
    SBInterpolatedKImage::SBInterpolatedKImageImpl::SBInterpolatedKImageImpl(
        const BaseImage<double>& data,
        double dk, double stepk, boost::shared_ptr<Interpolant2d> kInterp,
        double xcen, double ycen, bool cenIsSet, const GSParamsPtr& gsparams) :
    ...,
    _xcentroid(xcen), _ycentroid(ycen), _cenIsSet(cenIsSet)
{
    ...
    
    Nk = data.getYMax()-data.getYMin()+1)
    _ktab = boost::shared_ptr<KTable>(new KTable(Nk, _dk));
    std::complex<double>* kptr = _ktab.getArray();
    const double* ptr = data.getData();
    for(int i=Nk*Nk; i>0; --i, ptr+=2)
        *kptr++ = std::complex<double>(*ptr, *ptr+1);
}

ConstImageView<double> SBInterpolatedKImage::SBInterpolatedKImageImpl::getKData() const
{
    int Nk = _ktab->getN();
    data = reinterpret_cast<const double*>(_ktab->getArray())
    return ConstImageView<double>(data, boost::shared_ptr<double>(),
                                  2*N, Bounds<int>(0,2*N-1,0,N-1));
}
```
I didn't try this out, so there could be bugs in the above, but this should be close.